### PR TITLE
VZ 7406: Availability E2E tests

### PR DESCRIPTION
**Pull request by anders-swanson**
Implement an end to end test for availability.

### CODE REVIEW

**Review by michael-cico — APPROVED** [review]

File: tests/e2e/update/availability/availability_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	. "github.com/onsi/gomega"
+	"github.com/verrazzano/verrazzano/pkg/k8sutil"
+	"github.com/verrazzano/verrazzano/tests/e2e/pkg/test/framework"
+	"time"
+)
+
+const (
+	pollingInterval = 10 * time.Second
+	timeout         = 10 * time.Minute
+	deployName      = "rancher"
```
Comment (pfmackin) [inline tests/e2e/update/availability/availability_test.go:23]: are you going to test the other components?

Comment (anders-swanson) [reply tests/e2e/update/availability/availability_test.go:23]: Individual component's IsAvailable should be tested via unit tests. This test verifies that availability is dynamic and gets updated if a component is taken offline (in this case, all replicas are removed).

Comment (anders-swanson) [reply tests/e2e/update/availability/availability_test.go:23]: i.e., the unit tests should have all the individual component cases

Comment (pfmackin) [reply tests/e2e/update/availability/availability_test.go:23]: ok

**Pull request merged**